packages/ai/tests/ai/common/test_document_store_parallel.py

```
@@ -0,0 +1,473 @@
+Test philosophy:
+  - No real network I/O: all store calls are mocked with time.sleep to simulate
+    latency and prove parallelism via wall-clock timing assertions.
+  - All assertions are deterministic: result correctness is verified independent
+    of thread scheduling order.
+"""
+
+from __future__ import annotations
+
+import threading
+import time
+from typing import Any, Callable, Dict, List, Optional, Tuple
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)